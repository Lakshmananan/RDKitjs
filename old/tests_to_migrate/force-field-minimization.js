'use strict';

var RDKit = require('..');

beforeEach(() => RDKit.load());

test('MMFFoptimizeMolecule', function() {
  var smi = 'CCCCCOC(CO)';
  var mol = RDKit.Molecule.smilesToMol(smi);
  mol.addHs();
  mol.EmbedMolecule();
  mol.MMFFoptimizeMolecule();
  mol.removeHs();
  expect(mol.molToMolfile()).toMatchSnapshot();
  mol.delete();
});

test('MMFFoptimizeMolecule with paramaters', function() {
  var smi = 'CCCCCOC(CO)';
  var mol = RDKit.Molecule.smilesToMol(smi);
  mol.addHs();
  mol.EmbedMolecule();
  mol.MMFFoptimizeMoleculearg(1000, 'MMFF94');
  mol.removeHs();
  expect(mol.molToMolfile()).toMatchSnapshot();
  mol.delete();
});

test('MMFFOptimizeMoleculeConfs', function() {
  var smi = 'CCCCCOC(CO)';

  var mol = RDKit.Molecule.smilesToMol(smi);
  mol.addHs();
  mol.EmbedMultipleConfsarg(3, 1000, 2015);
  mol.MMFFOptimizeMoleculeConfs(8, 1000, 'MMFF94');
  mol.removeHs();
  expect(mol.sdwriteConfs()).toMatchSnapshot();
  mol.delete();
});

test('UFFOptimizeMolecule', function() {
  var smi = 'CCCCCOC(CO)';
  var mol = RDKit.Molecule.smilesToMol(smi);
  mol.addHs();
  mol.EmbedMolecule();
  mol.UFFOptimizeMolecule();
  mol.removeHs();
  expect(mol.molToMolfile()).toMatchSnapshot();
  mol.delete();
});

test('EmbedMolecule', function() {
  var smi = 'CCCCCOC(CO)';
  var mol = RDKit.Molecule.smilesToMol(smi);
  mol.addHs();
  mol.EmbedMoleculearg(1000, 1);
  mol.removeHs();
  expect(mol.molToMolfile()).toMatchSnapshot();
  mol.delete();
});
