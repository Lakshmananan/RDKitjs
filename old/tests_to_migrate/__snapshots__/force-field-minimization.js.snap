// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedMolecule 1`] = `
"
     RDKit          3D

  9  8  0  0  0  0  0  0  0  0999 V2000
   -3.4616    1.4289    0.4314 C   0  0  0  0  0  0  0  0  0  0  0  0
   -2.9477    0.0284    0.1317 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.4668    0.0949   -0.1176 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.8657   -1.2686    0.0740 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.6165   -1.2723   -0.1495 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.2728   -0.1862    0.4022 O   0  0  0  0  0  0  0  0  0  0  0  0
    2.6359   -0.3660    0.1169 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.3801    0.9281    0.0006 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.5567    0.7909    0.7537 O   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  0
  6  7  1  0
  7  8  1  0
  8  9  1  0
M  END
$$$$
"
`;

exports[`MMFFOptimizeMoleculeConfs 1`] = `
"
     RDKit          3D

  9  8  0  0  0  0  0  0  0  0999 V2000
    4.2049   -0.1343   -0.0700 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.7636    0.1674   -0.4518 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.8530    0.1802    0.7765 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.4134    0.5914    0.4625 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.3280   -0.4062   -0.4258 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.6681    0.0257   -0.6665 O   0  0  0  0  0  0  0  0  0  0  0  0
   -2.5743   -0.3641    0.3649 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.9713    0.1350    0.0157 C   0  0  0  0  0  0  0  0  0  0  0  0
   -4.8966   -0.2437    1.0272 O   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  0
  6  7  1  0
  7  8  1  0
  8  9  1  0
M  END
$$$$

     RDKit          3D

  9  8  0  0  0  0  0  0  0  0999 V2000
   -3.8544   -0.0601   -0.7140 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.0532    1.0931   -0.1298 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.7683    0.6501    0.5730 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.7396    0.0371   -0.3759 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.5628   -0.2933    0.3465 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.4641   -0.8645   -0.6010 O   0  0  0  0  0  0  0  0  0  0  0  0
    2.7225   -1.2189   -0.0276 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.6792   -0.0304    0.0174 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.9275   -0.4269    0.5719 O   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  0
  6  7  1  0
  7  8  1  0
  8  9  1  0
M  END
$$$$

     RDKit          3D

  9  8  0  0  0  0  0  0  0  0999 V2000
   -3.4103    0.9186   -0.5602 C   0  0  0  0  0  0  0  0  0  0  0  0
   -2.9693    0.3926    0.7969 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.9105   -0.7087    0.7105 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.5777   -0.2243    0.1392 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.4764   -1.3299    0.1717 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.6805   -0.9212   -0.4768 O   0  0  0  0  0  0  0  0  0  0  0  0
    2.5109   -0.0983    0.3386 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.8150    0.1867   -0.3998 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.6449    1.0642    0.3526 O   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  0
  6  7  1  0
  7  8  1  0
  8  9  1  0
M  END
$$$$
"
`;

exports[`MMFFoptimizeMolecule 1`] = `
"
     RDKit          3D

  9  8  0  0  0  0  0  0  0  0999 V2000
    3.1421    1.8386   -0.4923 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.6872    0.3939   -0.6248 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.2560    0.2090   -0.1205 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.8032   -1.2480   -0.2553 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.6112   -1.4612    0.2765 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.5285   -0.7132   -0.5206 O   0  0  0  0  0  0  0  0  0  0  0  0
   -2.8815   -0.8099   -0.0574 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.1425    0.2334    1.0272 C   0  0  0  0  0  0  0  0  0  0  0  0
   -2.8418    1.5355    0.5278 O   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  0
  6  7  1  0
  7  8  1  0
  8  9  1  0
M  END
$$$$
"
`;

exports[`MMFFoptimizeMolecule with paramaters 1`] = `
"
     RDKit          3D

  9  8  0  0  0  0  0  0  0  0999 V2000
   -3.8778    0.3120   -0.5134 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.0248   -0.5928    0.3621 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.6103   -0.0585    0.5998 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.7594   -0.0088   -0.6737 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.6675    0.4592   -0.3946 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.3211   -0.4950    0.4391 O   0  0  0  0  0  0  0  0  0  0  0  0
    2.6477   -0.1175    0.8000 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.6604   -0.4181   -0.3038 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.9785   -0.0672    0.1025 O   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  0
  6  7  1  0
  7  8  1  0
  8  9  1  0
M  END
$$$$
"
`;

exports[`UFFOptimizeMolecule 1`] = `
"
     RDKit          3D

  9  8  0  0  0  0  0  0  0  0999 V2000
   -3.2713    0.4667   -1.0876 C   0  0  0  0  0  0  0  0  0  0  0  0
   -2.9791   -0.8786   -0.4235 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.8749   -0.7887    0.6411 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.4844   -0.5591    0.0307 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.6043   -0.5961    1.1078 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.8803   -0.4433    0.5144 O   0  0  0  0  0  0  0  0  0  0  0  0
    2.3128    0.8990    0.6042 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.6179    1.0667   -0.1689 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.6397    0.3074    0.4191 O   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  0
  6  7  1  0
  7  8  1  0
  8  9  1  0
M  END
$$$$
"
`;
